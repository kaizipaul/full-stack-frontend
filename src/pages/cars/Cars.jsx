import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getCars } from '../../redux/cars/cars';
import './cars.css';
import CarCarousel from '../../components/carousel/CarCarousel';

const Cars = () => {
  const [currentUser, setCurrentUser] = useState('');
  const cars = useSelector((state) => state.cars);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getCars());
    setCurrentUser(localStorage.getItem('current_user'));
  }, [dispatch]);

  return (
    <div className="d-flex flex-column align-items-center justify-content-center w-100 overflow-hidden">
      <h3 className="user-name">{`Welcome, ${currentUser}`}</h3>
      <h2 className="cars-section-title">Latest Models</h2>
      <p style={{ color: 'rgb(182 183 184)' }} className="mb-0">Please select a car model</p>
      <div className="d-flex flex-row cars-container position-relative justify-content-around">
        <CarCarousel cars={cars} />
      </div>
    </div>
  );
};

export default Cars;
