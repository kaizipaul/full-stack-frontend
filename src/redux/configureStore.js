import { configureStore, combineReducers } from '@reduxjs/toolkit';
// import { carsReducer } from './cars/carsView';
import reservationReducer from './reservation/reserve';
import createCarReducer from './newCar/car';
import deleteCarReducer from './deleteCar/deleteCar';
import { authReducer } from './auth/auth';

const token = localStorage.getItem('token');
const initialState = {
  auth: {
    token: token || null,
    isAuthenticated: !!token,
  },
};

const rootReducer = combineReducers({
  // cars: carsReducer,
  newcar: createCarReducer,
  auth: authReducer,
  delete: deleteCarReducer,
  reservations: reservationReducer,
});

export default configureStore({
  reducer: rootReducer,
  preloadedState: initialState,
});
