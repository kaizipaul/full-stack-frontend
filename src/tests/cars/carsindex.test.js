import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureStore from '../../redux/configureStore';
import Cars from '../../pages/cars/Cars';

jest.mock('axios', () => ({
  get: jest.fn(() => Promise.resolve({ data: 'mock data' })),
}));

describe('All cars', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={configureStore}>
        <BrowserRouter>
          <Cars />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
