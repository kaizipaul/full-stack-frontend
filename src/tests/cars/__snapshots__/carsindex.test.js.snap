// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All cars renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex flex-column align-items-center justify-content-center w-100 overflow-hidden"
      >
        <h4
          class="user-name"
        >
          Welcome, null
        </h4>
        <h2
          class="cars-section-title"
        >
          Latest Models
        </h2>
        <p
          class="mb-0"
        >
          Please select a car model
        </p>
        <div
          class="d-flex flex-row cars-container position-relative justify-content-around"
        >
          <div
            class="d-flex justify-content-center mt-3"
          >
            <button
              class="pagination-btn btn left"
              disabled=""
              type="button"
            >
              <svg
                fill="none"
                height="3em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 15 15"
                width="3em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 4L9 11L4.5 7.5L9 4Z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
          <div
            class="w-100 carousel slide"
          >
            <div
              class="carousel-indicators"
            />
            <div
              class="carousel-inner"
            />
          </div>
          <div
            class="d-flex justify-content-center mt-3"
          >
            <button
              class="pagination-btn btn right"
              disabled=""
              type="button"
            >
              <svg
                fill="none"
                height="3em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 15 15"
                width="3em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M6 11L6 4L10.5 7.5L6 11Z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex flex-column align-items-center justify-content-center w-100 overflow-hidden"
    >
      <h4
        class="user-name"
      >
        Welcome, null
      </h4>
      <h2
        class="cars-section-title"
      >
        Latest Models
      </h2>
      <p
        class="mb-0"
      >
        Please select a car model
      </p>
      <div
        class="d-flex flex-row cars-container position-relative justify-content-around"
      >
        <div
          class="d-flex justify-content-center mt-3"
        >
          <button
            class="pagination-btn btn left"
            disabled=""
            type="button"
          >
            <svg
              fill="none"
              height="3em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 15 15"
              width="3em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 4L9 11L4.5 7.5L9 4Z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <div
          class="w-100 carousel slide"
        >
          <div
            class="carousel-indicators"
          />
          <div
            class="carousel-inner"
          />
        </div>
        <div
          class="d-flex justify-content-center mt-3"
        >
          <button
            class="pagination-btn btn right"
            disabled=""
            type="button"
          >
            <svg
              fill="none"
              height="3em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 15 15"
              width="3em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 11L6 4L10.5 7.5L6 11Z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
